Reduce list spacing

Vaishakh-SM opened the pull request:
**Give a summary of what the PR does, explaining any non-trivial design decisions**

This PR makes two CSS improvements:

1. Moves list-related CSS from index.css to the Markdown component.
2. Reduces list item spacing by adjusting the line height.

**Before**
![Screenshot from 2024-10-02 21-31-41](https://github.com/user-attachments/assets/1c1f1b29-2200-41f7-aa1e-a3287965b5d2)

**After**
![Screenshot from 2024-10-02 21-33-14](https://github.com/user-attachments/assets/8fe0ac31-8eda-49c5-9826-c62846c95ecc)


---
**Link of any specific issues this addresses**

Resolves #4082

tobitege commented:
This already looks much better, thanks for working on this!

Though not in the original issue, there's still quite a gap above/below the list (not sure which padding that is), but would you mind making that a little narrower, too, please?

Vaishakh-SM commented:
> Though not in the original issue, there's still quite a gap above/below the list (not sure which padding that is), but would you mind making that a little narrower, too, please?

Sure!

I also plan on addressing the lint errors related to this change.

Vaishakh-SM commented:
I've made the following changes:

- Removed an unused import from the file (was giving a lint error)
- Moved ul and ol components to a new file under the markdown directory (similar to the code file)
- Reduced the spacing between markdown elements.

It seems like the spacing was not caused by padding or the margin of the list, but rather by the spacing between children of the markdown. (Please correct me if I'm wrong here). Reducing the spacing here seemed aesthetically more pleasing.

![Screenshot from 2024-10-03 01-17-25](https://github.com/user-attachments/assets/ecdf89df-e857-4ec5-ab7c-ff0c48ef545b)

tobitege commented:
Thanks! Our UI specialist will have a final look once he's got time, so please bear with us. 👍

Vaishakh-SM commented on frontend/src/components/chat/ChatMessage.tsx:
```diff
@@ -89,7 +89,14 @@ function ChatMessage({
         </button>
       )}
-      <Markdown components={{ code }} remarkPlugins={[remarkGfm]}>
+      <Markdown
```
Oops, seems like I removed remarkGfm by mistake. Ill add this back in

rbren commented:
Looking much better! I might also shrink the space before and after the list a bit

Vaishakh-SM commented:
> Looking much better! I might also shrink the space before and after the list a bit

Cool!
I've reduced the spacing between before and after the list a bit more. From my understanding, this spacing is caused by the spacing between children of the Markdown component; I've reduced the spacing between the children to shrink the space between them. So this won't be specific to lists but to any children of the mardown component.

Please let me know if this is fine or if we need to follow a different approach here.

tobitege commented:
> Please let me know if this is fine or if we need to follow a different approach here.

May I ask for an example screenie of the newest version, please? 😃

Vaishakh-SM commented:
> May I ask for an example screenie of the newest version, please? 😃

Sure! 😃
![Screenshot from 2024-10-04 00-41-07](https://github.com/user-attachments/assets/c1e5bfdc-9ca1-4d73-a9f7-c91b92a02589)

tobitege commented:
Thanks! I don't see a big difference above/below the list compared to the previous image, though? 🤔 
Sorry for being picky 😄

Vaishakh-SM commented:
> Thanks! I don't see a big difference above/below the list compared to the previous image, though? 🤔 Sorry for being picky 😄

Does this look any better? 😄
![Screenshot from 2024-10-04 00-59-18](https://github.com/user-attachments/assets/a7e462a1-4c13-4054-93db-022169fa365f)

tobitege commented:
Yes! Personally, I like that last version best, awesome! Thanks!!
@rbren wdyt?

rbren reviewed (approved):
Great work!
